objects/aidBoard/script.lua

```diff
@@ -848,46 +848,72 @@ function toggleElements()
     end
 end
 
-function scanElements()
-    local elemCardTable = {}
-    for _,guid in pairs(Global.getVar("elementScanZones")) do
-        local zone = getObjectFromGUID(guid)
-        --Go through all items found in the zone
-        for _, entry in ipairs(zone.getObjects()) do
-            --Ignore non-cards
-            if entry.type == "Card" then
-                --Ignore if no elements entry
-                if entry.getVar("elements") ~= nil then
-                    if not entry.is_face_down and entry.getPosition().z > zone.getPosition().z then
-                        table.insert(elemCardTable, entry)
-                    end
-                end
-            end
+local Elements = {}
+Elements.__index = Elements
+function Elements:new(init)
+    local outTable = {0,0,0,0,0,0,0,0}
+    setmetatable(outTable, self)
+    outTable:add(init)
+    return outTable
+end
+function Elements:add(other)
+    if other == nil then
+        return
+    elseif type(other) == "table" then
+        for i = 1, 8 do
+            self[i] = self[i] + other[i]
+        end
+    elseif type(other) == "string" then
+        for i = 1, string.len(other) do
+            self[i] = self[i] + math.floor(string.sub(other, i, i))
         end
     end
-    local combinedElements = elemCombine(elemCardTable)
-    local elementsTable = {"Sun","Moon","Fire","Air","Water","Earth","Plant","Animal"}
-    for i,total in ipairs(combinedElements) do
-        getObjectFromGUID(elementGuids[i]).editButton({
-            index = 0,
-            label = total + #getObjectsWithTag(elementsTable[i]),
-        })
+end
+function Elements:scale(op)
+    for i = 1, 8 do
+        self[i] = self[i] * op
     end
 end
-function elemCombine(inTableOfElemStrCards)
-    local outTable = {0,0,0,0,0,0,0,0}
-    for i = 1, #inTableOfElemStrCards do
-        local elemTable = elemStrToArr(inTableOfElemStrCards[i].getVar("elements"))
-        for j = 1, 8 do
-            outTable[j] = outTable[j] + elemTable[j]
+function Elements:__tostring()
+    return table.concat(self, "")
+end
+
+function scanSeaTile()
+    local seaTile = getObjectFromGUID("seaTile")
+        local elements = Elements:new()
+
+    local bounds = seaTile.getBoundsNormalized().size
+    local hits = Physics.cast{
+        origin = seaTile.getPosition(),
+        size = Vector(bounds.x, 10, bounds.z),
+        direction = Vector(0,1,0),
+        max_distance = 0,
+        type = 3, -- box
+        --debug = true,
+    }
+    for _, hit in pairs(hits) do
+        if hit.hit_object.type == "Generic" then
+            if hit.hit_object.getVar("elements") ~= nil then
+                elements:add(hit.hit_object.getVar("elements"))
+            end
         end
     end
-    return outTable
+    elements:scale(2)
```

Comment:
this looks to be multiplying every element on sea tile by 2. but for the purposes of Aided by events, i know people sometimes put their payment elements onto the sea tile

Reply:
er i guess not the sea tile, but also not their player area